pkg/router/template/router.go

```diff
@@ -398,6 +399,12 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 	config := ServiceAliasConfig{
 		Host: host,
 		Path: route.Spec.Path,
+		Hash: backendKey,
+	}
+
+	// Possibly hash the name for later use in obsuring cookie ID
+	if os.Getenv("COOKIE_PREFIX_MAIN") + os.Getenv("COOKIE_PREFIX_EDGE") + os.Getenv("COOKIE_PREFIX_REENCRYPT") != "" {
```

Comment:
On 04/25/2016 02:20 PM, Ram Ranganathan wrote:
>
> In pkg/router/template/router.go 
> <https://github.com/openshift/origin/pull/8615#discussion_r60961740>:
>
> > @@ -398,6 +399,12 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
> >  	config := ServiceAliasConfig{
> >  		Host: host,
> >  		Path: route.Spec.Path,
> > +		Hash: backendKey,
> > +	}
> > +
> > +	// Possibly hash the name for later use in obsuring cookie ID
> > +	if os.Getenv("COOKIE_PREFIX_MAIN") + os.Getenv("COOKIE_PREFIX_EDGE") + os.Getenv("COOKIE_PREFIX_REENCRYPT") != "" {
>
> @pecameron <https://github.com/pecameron> as the @eparis 
> <https://github.com/eparis> commented and based on the later comments, 
> just set the hash to something like:
> |config.Hash = fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%s_%s", 
> tls.Termination, backendKey))))| unconditionally and then use it 
> |{{cfg.Hash}}|` as the cookie name in the template.
>
> That way you don't need any environment variables.
> And maybe use
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/openshift/origin/pull/8615/files/efbfa149db72f9378c1378793e9db0263996d6f0#r60961740>
>
Hi Ram,
Thanks!  How do I test this new feature in 
pkg/router/template/router_test.go?
Thanks,
phil